src/language-js/parser-typescript.js

```diff
@@ -30,12 +30,16 @@ function parse(text, parsers, opts) {
     }
   }
 
-  return postprocess(ast, { ...opts, originalText: text });
+  return postprocess(result.ast, {
+    ...opts,
+    originalText: text,
+    tsParseResult: result,
+  });
 }
 
 function tryParseTypeScript(text, jsx) {
   const parser = require("@typescript-eslint/typescript-estree");
-  return parser.parse(text, {
+  return parser.parseAndGenerateServices(text, {
```

Comment:
I wonder about the performance impact of this change.

Reply:
I haven't check, but I think they generate services anyway.

@bradzacher Can you tell us the difference?

Reply:
Maybe we can use parse if no `@` ?

Reply:
without providing `parserOptions.project` it shouldn't be too different I'd have thought.

If you call `parse`, we'll create a `sourceFile`, convert it and that's it.
If you call `parseAndGenerateServices`, we'll create a TS program with one file in isolated mode, parse it and convert it.

There'll be a small overhead as TS will create some data-structures behind the scenes to enable you to access type information - but as all type information is computed lazily it shouldn't be an issue.

We could always expose a new API like `parseWithNodeMaps` which is like `parse` except it returns the AST + node maps.

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/2760

I will merge this and it'll be available with our release next monday

Reply:
Great! Thank  you!  @bradzacher